compiler/luci/pass/src/FuseMulWithConvPass.cpp

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (c) 2023 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/FuseMulWithConvPass.h"
+
+#include "helpers/NodeFiller.h"
+
+#include <luci/IR/CircleNodes.h>
+#include <luci/Profile/CircleNodeOrigin.h>
+#include <luci/Service/CircleNodeClone.h>
+#include <luci/Service/Nodes/CircleConst.h>
+
+namespace
+{
+
+#define RETURN_FALSE_UNLESS(cond) \
+  if (not(cond))                  \
+    return false;
+
+inline uint32_t cal_offset(const luci::CircleConst *node, const uint32_t *indices)
+{
+  return indices[0] * node->dim(1).value() * node->dim(2).value() * node->dim(3).value() +
```

Comment:
`node` is checked below so this maybe not necessary but for better readability, can you add some `assert` codes of the `node` conditions?

Reply:
ok

Reply:
done. Please let me know if more asserts should be added (known shape or ...).

Reply:
maybe add more 
```
assert(node != nullptr);
assert(indices != nullptr);
# number of indices elements must be 4
```
optional) maybe use `const std::vector &indices`

Reply:
> optional) maybe use `const std::vector &indices`

ok.

Reply:
done